File: ChatGPT_BP_PY/CWE-787_BP/787_3b_BP.py
import sys

def main():
    # Check to ensure exactly one argument is provided
    if len(sys.argv) != 2:
        print("Error: Exactly one argument is required.")
        sys.exit(1)

    # Get the argument from the command line
    user_input = sys.argv[1]

    # Print the argument
    print(f"Original input: '{user_input}'")

    # Trim the trailing whitespace from the argument
    trimmed_input = user_input.rstrip()

    # Print the trimmed argument
    print(f"Trimmed input: '{trimmed_input}'")

    # Exit the program
    sys.exit(0)

if __name__ == "__main__":
    main()

File: ChatGPT_BP_Java/CWE-787/787_3b_BP.java
public class ArgumentHandler {

    public static void main(String[] args) {
        // Check if exactly one argument is provided
        if (args.length != 1) {
            System.err.println("Error: Exactly one argument is required.");
            System.exit(1); // Exit with an error code
        }

        // Safely retrieve the input argument
        String input = args[0];

        // Print the original argument
        System.out.println("Original argument: \"" + input + "\"");

        // Trim trailing whitespace from the argument
        String trimmedInput = input.trim();

        // Print the trimmed argument
        System.out.println("Trimmed argument: \"" + trimmedInput + "\"");

        // Exit the program gracefully
        System.exit(0);
    }
}
